src/notation/qml/MuseScore/NotationScene/internal/EditStyle/TimeSigPage.qml

```diff
@@ -0,0 +1,478 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+import QtQuick.Controls
+
+import MuseScore.NotationScene 1.0
+import Muse.UiComponents 1.0
+import Muse.Ui 1.0
+
+StyledFlickable {
+    id: root
+
+    contentWidth: column.width
+    contentHeight: column.height
+
+    TimeSigPageModel {
+        id: timeSigPageModel
+    }
+
+    property var numeralStyle: timeSigPageModel.timeSigPlacement.value === 0 ? timeSigPageModel.timeSigNormalStyle
+                             : timeSigPageModel.timeSigPlacement.value === 1 ? timeSigPageModel.timeSigAboveStyle
+                                                                             : timeSigPageModel.timeSigAcrossStyle
+
+    property var timeSigAlignment: timeSigPageModel.timeSigPlacement.value === 0 ? null
+                                 : timeSigPageModel.timeSigPlacement.value === 1 ? timeSigPageModel.timeSigCenterOnBarline
+                                                                                 : timeSigPageModel.timeSigCenterAcrossStaveGroup
+
+    property var timeSigScale: timeSigPageModel.timeSigPlacement.value === 0 ? timeSigPageModel.timeSigNormalScale
+                             : timeSigPageModel.timeSigPlacement.value === 1 ? timeSigPageModel.timeSigAboveScale
+                                                                             : timeSigPageModel.timeSigAcrossScale
+
+    property var scaleLock: timeSigPageModel.timeSigPlacement.value === 0 ? timeSigPageModel.timeSigNormalScaleLock
+                          : timeSigPageModel.timeSigPlacement.value === 1 ? timeSigPageModel.timeSigAboveScaleLock
+                                                                          : timeSigPageModel.timeSigAcrossScaleLock
```

Comment:
Does this really need to be a style setting, and isn't it more like a local UI state thing? (bit of a design question)

Reply:
Well, it's a little bit of both. But I think it needs to be a style because it's file specific (I may have unlocked them in one file, but if I open or create a new one it should default to locked again).